bindata/scripts/kargs.sh

```diff
@@ -9,8 +9,67 @@ args=$(chroot /host/ cat /proc/cmdline)
 
 IS_OS_UBUNTU=true; [[ "$(chroot /host/ grep -i ubuntu /etc/os-release -c)" == "0" ]] && IS_OS_UBUNTU=false
 
-# Kernel args configuration isn't supported for Ubuntu now, so we shouldn't do anything here
 if ${IS_OS_UBUNTU} ; then
+    grub_config="/etc/default/grub"
+    # Operate on the copy of the file
+    cp /host/${grub_config} /tmp/grub
+
+    for t in "${kargs[@]}";do
+        if [[ $command == "add" ]];then
+            # Modify only GRUB_CMDLINE_LINUX_DEFAULT line if it's not already present
+            line=$(grep -P "^\s*GRUB_CMDLINE_LINUX_DEFAULT" /tmp/grub)
+            if [ $? -ne 0 ];then
+                exit 1
+            fi
+
+            IFS='"' read g param q <<< "$line"
+            arr=($param)
+            found=false
+
+            for item in "${arr[@]}"; do
+                if [[ "$item" == "${t}" ]]; then
+                    found=true
+                    break
+                fi
+            done
+
+            if [ $found == false ];then
```

Comment:
what will happen if we have a karg already set with a certain value and we want to change to a different value ?

example:
current grub cmdline args have: `ib_core.netns_mode=1` then we want to set (using kargs.sh) `ib_core.netns_mode=0`

in this case we will end up with both set in grub cmdline IIUC since we compare the full key=val

i think we have the same issue with rpm-ostree as well as we use only append and delete.

imo we need to seach for the key of the karg first, if exists and its value is different then we remove and readd with new value (rpm-ostree has --replace flag)

Reply:
Let's address it in a follow-up PR both for Ubuntu and RHEL-based OSes